stock_cycle_count/models/stock_cycle_count.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2017-18 ForgeFlow S.L.
+#   (http://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockCycleCount(models.Model):
+    _name = "stock.cycle.count"
+    _description = "Stock Cycle Counts"
+    _inherit = "mail.thread"
+    _order = "id desc"
+
+    name = fields.Char(readonly=True)
+    location_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Location",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    responsible_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Assigned to",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        tracking=True,
+    )
+    date_deadline = fields.Date(
+        string="Required Date",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        tracking=True,
+    )
+    cycle_count_rule_id = fields.Many2one(
+        comodel_name="stock.cycle.count.rule",
+        string="Cycle count rule",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        tracking=True,
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Planned"),
+            ("open", "Execution"),
+            ("cancelled", "Cancelled"),
+            ("done", "Done"),
+        ],
+        default="draft",
+        tracking=True,
+    )
+    stock_adjustment_ids = fields.One2many(
+        comodel_name="stock.inventory",
+        inverse_name="cycle_count_id",
+        string="Inventory Adjustment",
+        tracking=True,
+    )
+    inventory_adj_count = fields.Integer(compute="_compute_inventory_adj_count")
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        required=True,
+        default=lambda self: self.env.company,
+        readonly=True,
+    )
+
+    @api.depends("stock_adjustment_ids")
+    def _compute_inventory_adj_count(self):
+        for rec in self:
+            rec.inventory_adj_count = len(rec.stock_adjustment_ids)
+
+    def do_cancel(self):
+        self.write({"state": "cancelled"})
+
+    def _prepare_inventory_adjustment(self):
+        self.ensure_one()
+        return {
+            "name": "INV/{}".format(self.name),
+            "cycle_count_id": self.id,
+            "location_ids": [(4, self.location_id.id)],
+            "exclude_sublocation": True,
+            "responsible_id": self.responsible_id.id,
+        }
+
+    @api.model
+    def create(self, vals):
+        vals["name"] = self.env["ir.sequence"].next_by_code("stock.cycle.count") or ""
+        return super(StockCycleCount, self).create(vals)
```

Comment:
You can adopt short notation
```suggestion
        return super().create(vals)
```